packages/core/src/tools/utils.spec.ts

```diff
@@ -317,3 +327,189 @@ describe('findCommaSeparatedValue', () => {
+
+  describe('source is an array', () => {
+    it('should create a new array if destination is undefined', () => {
+      const source = [1]
+      const result = mergeInto(undefined, source)
+      expect(result).not.toBe(source)
+      expect(result).toEqual(source)
+    })
+
+    it('should return the copy of source if the destination is not an array', () => {
+      const source = [1]
+      expect(mergeInto({}, source)).toEqual(source)
+    })
+
+    it('should mutate and return destination if it is an array', () => {
+      const destination = ['destination']
+      const source = ['source']
+      const result = mergeInto(destination, source)
+      expect(result).toBe(destination)
+      expect(result).toEqual(source)
+    })
+  })
+
+  describe('source is an object', () => {
+    it('should create a new object if destination is undefined', () => {
+      const source = {}
+      const result = mergeInto(undefined, source)
+      expect(result).not.toBe(source)
+      expect(result).toEqual(source)
+    })
+
+    it('should return the copy of source if the destination is not an object', () => {
+      const source = { a: 1 }
+      expect(mergeInto([], source)).toEqual(source)
+    })
+
+    it('should mutate and return destination if it is an object', () => {
+      const destination = {}
+      const source = { a: 'b' }
+      const result = mergeInto(destination, source)
+      expect(result).toBe(destination as any)
+      expect(result).toEqual(source)
+    })
+  })
+})
+
+describe('combine', () => {
+  it('should deeply add and replace keys', () => {
+    const sourceA = { a: { b: 'toBeReplaced', c: 'source a' } }
+    const sourceB = { a: { b: 'replaced', d: 'source b' } }
+    expect(combine(sourceA, sourceB)).toEqual({ a: { b: 'replaced', c: 'source a', d: 'source b' } })
+  })
+
+  it('should not replace with undefined', () => {
+    expect(combine({ a: 1 }, { a: undefined as number | undefined })).toEqual({ a: 1 })
+  })
+
+  it('should replace a sub-value with null', () => {
+    expect(combine({ a: {} }, { a: null as any })).toEqual({ a: null })
+  })
+
+  it('should ignore null arguments', () => {
+    expect(combine({ a: 1 }, null)).toEqual({ a: 1 })
+  })
+
+  it('should merge arrays', () => {
+    const sourceA = [{ a: 'source a' }, 'extraString'] as any
+    const sourceB = [{ b: 'source b' }] as any
+    expect(combine(sourceA, sourceB)).toEqual([{ a: 'source a', b: 'source b' }, 'extraString'])
+  })
+
+  it('should merge multiple objects', () => {
+    expect(combine({ a: 1 }, { b: 2 }, { c: 3 })).toEqual({ a: 1, b: 2, c: 3 })
+  })
+
+  it('should not keep references on objects', () => {
+    const source = { a: { b: 1 } }
+    const result = combine({}, source)
+    expect(result.a).not.toBe(source.a)
+  })
+
+  it('should not keep references on arrays', () => {
+    const source = { a: [1] }
+    const result = combine({}, source)
+    expect(result.a).not.toBe(source.a)
+  })
+})
+
+describe('deepClone', () => {
```

Comment:
how does it behave for other types of objects?
like function for example

Reply:
For functions, we don't clone - we just copy a reference. It's tested a few lines below (line 440). 
This implementation support all types that were supported by the previous implementation. There will be no breaking change.

That means:
 - we don't support `Set` and `Map` - I didn't add support for these types because we don't use them, and it would increase bundle size. 
 - we don't support custom classes - we don't clone prototype chain as it's non-trivial, and it seems we don't use it

Reply:
ok, got it.
as mentioned in https://github.com/DataDog/browser-sdk/pull/897/files#r663744326, it could be nice to have a more expressive test for this one, maybe also splitting primitive handling from other object types.

Reply:
sure, I will update test descriptions :)